### PR TITLE
additional lineplot works

### PR DESCRIPTION
**Pull request by moontrip**
This will probably rely on https://github.com/VEuPathDB/web-components/pull/333
This PR will be used for addressing additional lineplot works. Thus branched out from `lineplot-categorical` branch. 
Currently, this branch is up-to-date with main branch

### CODE REVIEW

**Comment by moontrip** [comment]
As a first commit, #1003 for lineplot Viz was addressed here. 
Next, axis range control will be implemented here.

**Comment by moontrip** [comment]
@bobular I have done several works for lineplot Viz. This branch/PR is up-to-date to `main` branch but for now, it is supposed to be merged into your branch, `lineplot-categorical`. I think that this PR may be used for UX discussion on Monday if they want to discuss lineplot - last week they cancelled it due to your vacation. FYI, I am out of VEuPathDB on Monday but may check email and slack, if not on time.

This PR contains several updates and relies on [web-components PR](https://github.com/VEuPathDB/web-components/pull/333) (please use yarn link):

- axis range control is now working: see the second and third screenshots below.
- I think that we can't support X-axis range control for categorical variable, so I introduced a new prop, `disabled`, for both (number/date) input form and (reset) button (via web-components' control and widgets) to disable. For that case, I also added to show text at title, like `Range (not available)`. Please see the first screenshot.
- The bug you found, switching number to date variable or vice versa (#1003) is also addressed
- For your information, I have made two methods for initial Y-axis range/viewport: Method 1 - considering variable's metadata for min/max; Method 2 - purely based on data min/max, which may be in line with Danica's thought. I left comment so you can test both by commenting out each of them: currently Method 1 is used (Method 2 is commented out)


### Screenshots
- categorical X-axis variable (X-axis range is disabled)
lineplot-xCategorical

- axis range controls and truncations for number variables
lineplot-truncations-numbers

- axis range controls and truncations for date variables
lineplot-truncations-dates

**Review by bobular — CHANGES_REQUESTED** [review]
Axis controls work great and code looks good.

Please take a look at the scatterplot number-to-date variable bug that still remains.

File: src/lib/core/components/visualizations/implementations/LineplotVisualization.tsx [hunk 1]
```diff
@@ -573,12 +573,31 @@ function LineplotViz(props: VisualizationProps) {
               : undefined,
           max: yMinMaxRange?.max,
         } as NumberOrDateRange)
-      : numberDateDefaultDependentAxisRange(
+      : // add conditions to prevent previous yMinMaxRange from calling the function, e.g., date to categorical
```
[the remark this answers is not in the document]
Comment (bobular) [reply src/lib/core/components/visualizations/implementations/LineplotVisualization.tsx:621]: PS. I haven't tested the alternatives for y-axis range calculations. We can wait for UX meeting feedback.

PPS. I did test the disabling of x axis range controls when ordinal (e.g. "age group") variables are used and that works great.

[the remark this answers is not in the document]
Comment (bobular) [reply src/lib/core/components/visualizations/implementations/LineplotVisualization.tsx:621]: Oh, I see scatterplot is fixed in https://github.com/VEuPathDB/web-eda/pull/1004

**Comment by bobular** [comment]
I made the viewport part of the data request use the `independentAxisRange`

It works fine, but every time the first marker is cut in half by the yellow warning shading.  Not sure of the best solution.
image

**Comment by moontrip** [comment]
@bobular I hope you enjoyed your relatively long vacation 😄 
And thank you for your reviews on my several PRs 👍 I am pleased that most of them worked fine. Yes scatter plot bug on the switch of number to date variable (or vice versa) was addressed in the separate PR. Speaking of your suggestion on the combined condition at axisRangeMargin() for both scatterplot and lineplot Vizs, I thought of it but didn't do that because lineplot Viz required one additional check condition for ordinal/categorical case.  

And thank you for adding the viewport feature for lineplot 👍: I was not sure if I should have made it for lineplot. Yes, I also noticed that the error bar often cut in half depending on the range input value but I did/do not have clear answer to address it either 🤔 Maybe it is acceptable? 😉

**Comment by bobular** [comment]
Danica and others said the cut in half points/error bars wasn't a show stopper yesterday.

**Review by bobular — APPROVED** [review]

**Pull request merged**